### ISSUE TITLE
Everyboard in boards.txt overides default compiler.c.flags, and compiler.cpp.flags

### ISSUE THREAD

**Issue by ricklon** [chipKIT32/chipKIT-core#67]
Every board in boards.txt overrides default compiler.c.flags, and compiler.cpp.flags even when they might not need to.

**Comment by EmbeddedMan** [comment]
What's the task to do here? To go through each compiler.cpp.flags assignment in every variant and remove it if it's the same as he master copy in boards.txt?

**Comment by ricklon** [comment]
Correct. If the board relies in default settings or needs custom settings.

--Rick

**Comment by EmbeddedMan** [comment]
So for an option like this, in an individual board.txt file, there's now
way to specify "keep the existing value from the global boards.txt file,
but add my own little customization"? That's too bad.

I think I can take care of this one too.

*Brian

**Comment by ricklon** [comment]
I think there is an equivalent of append, but I need to confirm Arduino
implemented it.

-_Rick

On Thu, Sep 17, 2015 at 10:46 AM, Brian Schmalz <notifications@github.com>
wrote:

> So for an option like this, in an individual board.txt file, there's now
> way to specify "keep the existing value from the global boards.txt file,
> but add my own little customization"? That's too bad.
>
> I think I can take care of this one too.
>
> *Brian
>
> On Thu, Sep 17, 2015 at 9:43 AM, Ricklon <notifications@github.com> wrote:
>
> > Correct. If the board relies in default settings or needs custom
> settings.
> >
> > --Rick
> >
> > On Thu, Sep 17, 2015, 10:08 AM Brian Schmalz <notifications@github.com>
> > wrote:
> >
> > > What's the task to do here? To go through each compiler.cpp.flags
> > > assignment in every variant and remove it if it's the same as he master
> > > copy in boards.txt?
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141098218
> > >
> >
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141108833
> >
> > .
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141110198>
> .
>



-- 
Co-founder
Fair Use Building and Research (FUBAR) Labs
http://fubarlabs.org

**Comment by EmbeddedMan** [comment]
Rick, I'm a little confused as to how this is supposed to work.

We currently have compiler.c.flags being set in
platform.txt
platforms.txt
boards.txt
and in each of the boards.txt files that is in each variant folder.

Which of these does Arduino IDE actually use?

*Brian

**Comment by ricklon** [comment]
/pic32
platform.txt
boards.txt

Arduino doesn't currently look in the variants folder. They use the
singular and MPIDE uses the plural.

--Rick

**Comment by EmbeddedMan** [comment]
Rick,

OK, that makes sense. The problem is that every board in boards.txt has a
different value for compiler.c.flags than the value in platform.txt.

So maybe the right thing to do is for me to update the value that's in
platform.txt, then remove all of the (now redundant) values from
boards.txt. Does that make sense?

*Brian

**Comment by ricklon** [comment]
That makes the most sense. But just do a test before converting everything.

We might need to file an issue with Arduino that their code doesn't append.

--Rick

On Thu, Sep 17, 2015 at 5:17 PM, Brian Schmalz <notifications@github.com>
wrote:

> Rick,
>
> OK, that makes sense. The problem is that every board in boards.txt has a
> different value for compiler.c.flags than the value in platform.txt.
>
> So maybe the right thing to do is for me to update the value that's in
> platform.txt, then remove all of the (now redundant) values from
> boards.txt. Does that make sense?
>
> *Brian
>
>
> On Thu, Sep 17, 2015 at 4:08 PM, Ricklon <notifications@github.com> wrote:
>
> > /pic32
> > platform.txt
> > boards.txt
> >
> > Arduino doesn't currently look in the variants folder. They use the
> > singular and MPIDE uses the plural.
> >
> > --Rick
> >
> > On Thu, Sep 17, 2015, 5:01 PM Brian Schmalz <notifications@github.com>
> > wrote:
> >
> >
> > > Rick, I'm a little confused as to how this is supposed to work.
> > >
> > > We currently have compiler.c.flags being set in
> > > platform.txt
> > > platforms.txt
> > > boards.txt
> > > and in each of the boards.txt files that is in each variant folder.
> > >
> > > Which of these does Arduino IDE actually use?
> > >
> > > *Brian
> > >
> > > On Thu, Sep 17, 2015 at 10:42 AM, Ricklon <notifications@github.com>
> > > wrote:
> > >
> > > > I think there is an equivalent of append, but I need to confirm
> Arduino
> > > > implemented it.
> > > >
> > > > -_Rick
> > > >
> > > > On Thu, Sep 17, 2015 at 10:46 AM, Brian Schmalz <
> > > notifications@github.com>
> > > > wrote:
> > > >
> > > > > So for an option like this, in an individual board.txt file,
> there's
> > > now
> > > > > way to specify "keep the existing value from the global boards.txt
> > > file,
> > > > > but add my own little customization"? That's too bad.
> > > > >
> > > > > I think I can take care of this one too.
> > > > >
> > > > > *Brian
> > > > >
> > > > > On Thu, Sep 17, 2015 at 9:43 AM, Ricklon <notifications@github.com
> >
> > > > wrote:
> > > > >
> > > > > > Correct. If the board relies in default settings or needs custom
> > > > > settings.
> > > > > >
> > > > > > --Rick
> > > > > >
> > > > > > On Thu, Sep 17, 2015, 10:08 AM Brian Schmalz <
> > > notifications@github.com
> > > > >
> > > > > > wrote:
> > > > > >
> > > > > > > What's the task to do here? To go through each
> compiler.cpp.flags
> > > > > > > assignment in every variant and remove it if it's the same as
> he
> > > > master
> > > > > > > copy in boards.txt?
> > > > > > >
> > > > > > > —
> > > > > > > Reply to this email directly or view it on GitHub
> > > > > > > <
> > > > > >
> > > > >
> > > >
> > >
> >
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141098218
> > > > > > >
> > > > > >
> > > > > > > .
> > > > > > >
> > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > <
> > > > >
> > > >
> > >
> >
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141108833
> > > > > >
> > > > > > .
> > > > >
> > > > > >
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > >
> > >
> >
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141110198
> > > > >
> > > > > .
> > > > >
> > > >
> > > >
> > > >
> > > > --
> > > > Co-founder
> > > > Fair Use Building and Research (FUBAR) Labs
> > > > http://fubarlabs.org
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141127187
> > > >
> > > > .
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141227448
> > >
> >
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141228823
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chipKIT32/chipKIT-core/issues/67#issuecomment-141230578>
> .
>



-- 
Co-founder
Fair Use Building and Research (FUBAR) Labs
http://fubarlabs.org

**Comment by EmbeddedMan** [comment]
How is the append feature supposed to work? If I understand that, I can
test it.

*Brian

**Comment by EmbeddedMan** [comment]
I believe this issue is now fixed. Please confirm.
